app/services/peoplesoft_service.rb

```diff
@@ -2,63 +2,87 @@
 
 # Service for applying registrar based actions from Peoplesoft
 class PeoplesoftService
-  def initialize(submission:)
-    @submission = submission
+  def self.update(...)
+    new(...).update
   end
 
-  # Assigns readers and records a new reader action for the submission
-  def new_reader_action(readers:, reader_action_attributes:)
-    assign_readers(readers)
-    @previous_action = submission.last_reader_action_at
-    @new_action = reader_action_attributes[:last_reader_action_at]
-
-    return unless valid_action?
-
-    submission.update!(reader_action_attributes)
-
-    return unless /reject/i.match?(submission.readerapproval)
-
-    submission.update!(rejection_attributes)
+  def initialize(submission:, submission_params:)
+    @submission = submission
+    @submission_params = submission_params
   end
 
-  # Records a new registrar action for the submission
-  def new_registrar_action(registrar_action_attributes:)
-    @previous_action = submission.last_registrar_action_at
-    @new_action = registrar_action_attributes[:last_registrar_action_at]
-
-    return unless valid_action?
-
-    submission.update!(registrar_action_attributes)
-
-    return complete_approval if submission.approved?
-
-    submission.update!(rejection_attributes)
+  def update # rubocop:disable Metrics/AbcSize
+    submission.transaction do
+      # First, assign the readers anew
+      submission.readers.destroy_all
+      submission.readers.create!(readers)
+
+      update_reader_actions!
+      update_registrar_actions!
+
+      # Handle when submission is rejected and when is has been fully approved
+      if /reject/i.match?(submission.readerapproval) || /reject/i.match?(submission.regapproval)
+        submission.update!(submitted_at: nil, submitted_to_registrar: 'false')
+      elsif submission.ready_for_cataloging?
+        CreateStubMarcRecordJob.perform_later(submission.druid)
+        CreateEmbargo.call(submission.druid, submission.embargo_release_date)
+      end
+    end
   end
 
   private
 
-  attr_reader :submission, :previous_action, :new_action
+  attr_reader :submission, :submission_params
 
-  # Assigns or updates readers for a submission
-  def assign_readers(readers)
-    submission.readers.destroy_all
-    submission.readers.create!(readers)
+  def readers
+    @readers ||= Reader.sorted_list(submission_params[:reader])
   end
 
-  def complete_approval
-    CreateStubMarcRecordJob.perform_later(submission.druid)
-    CreateEmbargo.call(submission.druid, submission.embargo_release_date)
+  def update_reader_actions!
+    return unless valid_action?(
+      new_action: reader_action_datetime,
+      previous_action: submission.last_reader_action_at
+    )
+
+    submission.update!(
+      readerapproval: submission_params[:readerapproval],
+      readercomment: submission_params[:readercomment],
+      last_reader_action_at: reader_action_datetime
+    )
   end
```

Comment:
Now the logic of the reader-related method and the registrar-related method is the same, which is how the old registrar data importer worked, I believe.